.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,233 @@
+          echo "NUM_OF_CPUS=$(nproc --all)" >> $GITHUB_ENV;
+      - uses: actions/checkout@v2
+      - name: Log in to the GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Extract metadata (tags, labels) for Docker
+        id: docker-image-metadata
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ghcr.io/${{ github.repository }}-${{ env.IMAGE_SUFFIX }}
+      # TODO: Could we just build in `docker-compose` build/context without publishing?
+      - name: Build and Push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          context: integration-test-apps/${{ matrix.instrumentation-type}}-instrumentation/${{ matrix.app-platform }}
+          tags: ${{ steps.docker-image-metadata.outputs.tags }}
+          labels: ${{ steps.docker-image-metadata.outputs.labels }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Prepare Metric Data Queries JSON Inputs
+        run: >-
+          for metric_kind in cpu-load total-memory;
+          do
+            echo "$(
+              cat ./.github/aws-cloudwatch-api-values/get-metric-data/$metric_kind-metric-data-query.json |
+              sed -e "s/<METRIC_PERIOD>/${{ env.HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS }}/g" \
+                  -e "s/<NUM_OF_CPUS>/$NUM_OF_CPUS/g" \
+                  -e "s/<LOGS_NAMESPACE>/$(echo ${{ env.LOGS_NAMESPACE }} | sed 's/\//\\\//g')/g" \
+                  -e "s/<PROCESS_COMMAND_LINE_DIMENSION_VALUE>/$(echo ${{ env.PROCESS_COMMAND_LINE }} | sed 's/\//\\\//g')/g"
+            )" |
+            tee ./.github/aws-cloudwatch-api-values/get-metric-data/prepared-$metric_kind-metric-data-query.json;
+          done
+      - name: Create Soak Test alarms
+        run: >-
+          declare -a COMMON_API_PARAMETERS=(
+          --evaluation-periods 5
+          --datapoints-to-alarm 3
+          --comparison-operator GreaterThanOrEqualToThreshold
+          --treat-missing-data ignore
+          );
+          aws cloudwatch put-metric-alarm "${COMMON_API_PARAMETERS[@]}" \
```

Comment:
Prefer to do stuff in Python, like we do in lambda

https://github.com/aws-observability/aws-otel-lambda/blob/main/adot/utils/soak/soak.py#L123

Reply:
I'll put the `aws cloudwatch put-metric-alarm` and `aws cloudwatch get-metric-data` calls into the `poll-during-soak-tests.py` script and use `boto3`. I liked how this info was immediately obvious from this `soak-testing.yml` file, but I can see how a python script is cleaner.

I'll keep the `.github/aws-cloudwatch-api-values/get-metric-data/cpu-load-metric-data-query.json` json template files if that's okay with you guys? I like how that file has the inputs formatted and how we have syntax highlighting there, but we can just make them python strings in `poll-during-soak-tests.py` if we think we are separating them out into too many files.

Reply:
For `aws cloudwatch get-metric-widget` I think it makes sense to continue using `awscli` because the goal is to show that these Soak Tests could be swapped out with another vendor rather simply. We use AWS alarms and "snapshots" for now, but the message I was hoping to convey was all we need to do is switch some `run:` commands and we could be using a different vendor no problem.

We decided using _a vendor_ was preferable because we get alarming and metric querying done for us, so the `boto3` in the `poll-during-soak-tests.py` script was a small necessity but even that should be easy to swap with another vendor if needed in the future.

Reply:
When I started moving the API calls to `poll-during-performance-tests.py` (I renamed it slightly), I realized a major downside to deleting the `.json` files; because `put_metric_alarm` and `get_metric_data` share _the same_ parameters for the Metric Query, we now have to _duplicate those API parameters_: one copy is in `poll-during-performance-tests.py` to create the alarms, and one is in a step of `soak-testing.yml` to request the metrics when it prepares the JSON for the graph visualization after the Soak Tests.

I think having the same Python dict (it's an array of dicts actually) in 2 different places is a big enough downside that the `.json` solution would be preferred. But I will convert `aws cloudwatch get-metric-data` to a Python script and so we can evaluate it plainly.

Reply:
Not sure I quite follow but it sounds like it could be solved by importing a shared `.py` file?

Reply:
Sorry if I'm not being clear, but importing a shared `.py` file would still be tricky. One script is run as a Docker Container alongside the others in the `docker-compose` file. So whatever `.py` file we make would still need to be copied over when the container is built. I don't think that is particularly easy to understand and copying and pasting the Python Dictionary is probably better at that point.

If it still doesn't makes sense, I'll have the updated code pushed to this PR in a few hours which should hopefully be easier to understand.